src/utils.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Pads the front of the given hex string with zeroes until it reaches the
+ * target length. If the input string is already longer than or equal to the
+ * target length, it is returned unmodified.
+ *
+ * If the input string is "0x"-prefixed or not a hex string, an error will be
+ * thrown.
+ *
+ * @param hexString The hexadecimal string to pad with zeroes.
+ * @param targetLength The target length of the hexadecimal string.
+ * @returns The input string front-padded with zeroes, or the original string
+ * if it was already greater than or equal to to the target length.
+ */
+export function padWithZeroes(hexString: string, targetLength: number): string {
+  if (hexString !== '' && !/^[a-f0-9]+$/iu.test(hexString)) {
+    throw new Error(
+      `Expected an unprefixed hex string. Received: ${hexString}`,
```

Comment:
The old function didn't include this check. It would coerce anything it was given into a string. I'm pretty confident it was only ever called with a hex string though, because in the one place it's called, it's given the result of a `Buffer.prototype.toString('hex')` call. So I'd guess this doesn't change behaviour.

Reply:
Yeah, my thinking was exactly that it's non-breaking for current usage, and we won't accidentally shoot ourselves in the foot in the future.